packages/wallet-sdk/src/CoinbaseWalletSDK.ts

```diff
@@ -25,7 +25,20 @@ export class CoinbaseWalletSDK {
 
   public makeWeb3Provider(preference: Preference = { options: 'all' }): ProviderInterface {
     const params = { metadata: this.metadata, preference };
-    return getCoinbaseInjectedProvider(params) ?? new CoinbaseWalletProvider(params);
+    let ethereum = undefined;
+    try {
+      const window = globalThis as CBWindow;
+      ethereum = window.ethereum ?? window.top?.ethereum;
+    } catch {
+      // do nothing, ethereum remains undefined
+    }
+
+    let injectedCbWalletMobileBrowserProvider = undefined;
+
+    if (ethereum?.isCoinbaseBrowser) {
+      injectedCbWalletMobileBrowserProvider = ethereum;
+    }
+    return injectedCbWalletMobileBrowserProvider ?? new CoinbaseWalletProvider(params);
```

Comment:
you may be able to simplify this logic further. i don't think we need to define ethereum and injectedCbWalletMobileBrowserProvider. some of the logic can be pulled out as helper method. 
it's just to check if you are in mobile browser. that should be all.

Reply:
cleaned it up a bit here: [ce01c48](https://github.com/coinbase/coinbase-wallet-sdk/pull/1332/commits/ce01c4843ad372de91b6467fa56d35ac20d8293e)